AtRuleFeatureDetection/explainer.md

```diff
@@ -0,0 +1,278 @@
+
+1. In the [Blink intent-to-ship thread for `@property`](https://groups.google.com/a/chromium.org/g/blink-dev/c/3ygpsew53a0/m/Ar_OPlthAwAJ), it was pointed out that authors need a mechanism to detect support so that they can fall back to `CSS.registerProperty()` if needed.
+2. A [StackOverflow question](https://stackoverflow.com/questions/44244221/is-it-possible-to-do-a-css-supports-check-on-a-media-rule) asks whether it is possible to detect support for `@media` features, for example to detect if the user agent can return a yes/no answer for `@media (pointer)`.
+3. A [Mastodon post](https://mastodon.social/@xro/113106213499516093) asks whether it is possible to test for style query support.
+4. At time of writing, several in-development CSS features propose to implement new at-rules. These include [`@sheet`](https://github.com/MicrosoftEdge/MSEdgeExplainers/blob/main/AtSheet/explainer.md) as well as [CSS Functions and Mixins](https://css.oddbird.net/sasslike/mixins-functions/).
+
+### Goals
+
+Allow authors to feature-detect newly introduced at-rules.
+
+Allow authors to feature-detect new enhancements to existing at-rules, such as:
+- New media query features and other additions to at-rule preludes
+- New descriptors that may be introduced to rules such as `@font-face`
+
+At-rule feature detection should be available in all contexts where CSS allows conditioning based on support
+of a feature. This includes, but is not limited to,
+`@supports`, `CSS.supports()`, `@import ... supports()`, and `@when supports()`.
+
+### Non-goals
+
+The CSS `@charset` rule, despite its appearance, is
+[not an at-rule](https://drafts.csswg.org/css-syntax/#charset-rule).
+Rather, `@charset` is a marker that can appear only as the first few bytes of a stylesheet file. It signals to
+the user agent what character encoding should be used to decode the contents of the stylesheet.
+
+CSS feature detection has, since its inception, relied on the test of "does it parse successfully?" to
+answer the question of whether a property-value pair is supported. Because of its special nature, user agents
+may have a different parsing implementation for `@charset` compared to true at-rules, which might not be as easily
+reused for feature detection.
+
+At the same time, there is far less of a use case for feature-detecting `@charset` compared to true at-rules.
+`@charset` is part of the [Baseline](https://developer.mozilla.org/en-US/docs/Web/CSS/@charset) feature set long 
+supported in all major browsers. On the modern Web, the encouraged solution for character encoding issues in 
+CSS is to use UTF-8.
+
+Accordingly, this explainer does not propose making `@charset` feature-detectable using `at-rule()`.
+
+## Proposed Approach
+
+The `at-rule()` function can be used in the following ways:
+
+### Detect whether an at-rule name is recognized at all
+
+This is the simplest form. An at-rule name is passed to the function; the result is true if the implementation

```

Comment:
I'd suggest expanding on the first sentence a bit. Maybe "In its simplest form, the `at-rule()` function can be used to..."